### ISSUE TITLE
On-Premise error 'Docs not built'

### ISSUE THREAD

**Issue by ruzy** [getsentry/sentry#2664]
I am running Sentry 8.1.1 on-premise with a machine that has no access to the internet.  When I hit the  project settings - Setup - Instructions page, the client hangs and an error is logged.

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/sentry/api/base.py", line 81, in handle_exception
    return super(Endpoint, self).handle_exception(exc)
  File "/usr/lib/python2.7/site-packages/sentry/api/base.py", line 145, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/sentry/api/endpoints/project_docs.py", line 14, in get
    raise RuntimeError('Docs not built')
RuntimeError: Docs not built
```

Do I need to install the Sentry docs manually or am I missing something from the configuration?

**Comment by mattrobenolt** [comment]
If you installed from pypi with `pip install sentry`, these should be included. Did you try to build from source?

**Comment by ruzy** [comment]
I did install from pypi using the `pip install` command.

**Comment by mattrobenolt** [comment]
Hmm, I wonder if we've regressed here then. I'll investigate.

**Comment by mattrobenolt** [comment]
I just tested from our compiled package, and they are definitely included on pypi.

Can you do this: Fire up `$ sentry shell`

Then enter the following:

```python
>>> from sentry.utils.integrationdocs import DOC_FOLDER, load_doc
>>> print(DOC_FOLDER)
```

Then `ls` on what `DOC_FOLDER` value is.

**Comment by mattrobenolt** [comment]
In your case, `DOC_FOLDER` _should_ be `/usr/lib/python2.7/site-packages/sentry/integration-docs`

**Comment by ruzy** [comment]
Correct, my `DOC_FOLDER` is `/usr/lib/python2.7/site-packages/sentry/integration-docs` however that folder does not exist in my install.  I've also run the upgrade to 8.1.2 with no luck in changes either.

**Comment by ruzy** [comment]
If I run `sentry repair` I can see that it attempts a documentation sync which fails.  In the case of this system not having access to the internet, it can't get updated docs, right?

**Comment by mattrobenolt** [comment]
Was this an upgrade from 7.x or something? I'm wondering if we're not dropping in the integration-docs folder somehow when it previously didn't exist. The 8.1.x packages definitely include these files.

**Comment by mattrobenolt** [comment]
Can you also dump output for `$ pip freeze` ?

**Comment by ruzy** [comment]
My initial install was 8.0.2.  As I mentioned earlier, the machine running Sentry has no internet access, so I downloaded the tar from pypi, copied it over to the machine and ran `pip install sentry-8.0.2.tar.gz' to install Sentry locally.  I've noticed that setting up the integration docs from `sentry/src/sentry/utils/integrationdocs.py' in the GitHub source `BASE_URL` requires internet access.  Could this be the underlying culprit?  Is there a way to side-load the docs?

**Comment by mattrobenolt** [comment]
Hmm, could you try from the whl file instead? https://pypi.python.org/packages/py27/s/sentry/sentry-8.1.2-py27-none-any.whl

Then `pip install 8.1.2-py27-non-any.whl` would be the same process. I think it's possible that the .tar.gz isn't doing it right because of our `MANIFEST.in`, so will need to investigate this. But it should be fine with the whl.

**Comment by mattrobenolt** [comment]
Just confirmed, the .tar.gz is broken in this regard, but the whl is fine. Will get this fixed up.

**Comment by ruzy** [comment]
Thank you!  It is working great now.

**Comment by mwcz** [comment]
I just performed the upgrade to 8.1.2 a few minutes ago and I'm seeing this error.  Is the tar.gz in PyPi still unhappy?

**Comment by DaveBlooman** [comment]
I am getting this error when using pip, version 8.1.2 @mattrobenolt

**Comment by mattrobenolt** [comment]
Yeah, this was broken, as mentioned in this ticket, for 8.1.2. It'll be fixed for 8.1.3. The problem is specifically in the .tar.gz distribution, but the .whl is fine.

**Comment by mwcz** [comment]
Ahhh, thanks @mattrobenolt!  I thought you might have fixed the 8.1.2 tar.gz but maybe pypi doesn't allow that.  No worries!

**Comment by mattrobenolt** [comment]
Nope, releases are immutable there. :)
